diff -up qtbase-everywhere-src-6.5.0-beta3/cmake/QtFlagHandlingHelpers.cmake.omv~ qtbase-everywhere-src-6.5.0-beta3/cmake/QtFlagHandlingHelpers.cmake
--- qtbase-everywhere-src-6.5.0-beta3/cmake/QtFlagHandlingHelpers.cmake.omv~	2023-02-25 11:43:05.707706181 +0100
+++ qtbase-everywhere-src-6.5.0-beta3/cmake/QtFlagHandlingHelpers.cmake	2023-02-25 11:43:49.746707525 +0100
@@ -40,6 +40,7 @@ function(qt_internal_add_linker_version_
         endif()
         string(APPEND contents "${current} { *; };\n")
 
+	if("${target}" STREQUAL "Core")
         foreach(minor_version RANGE ${PROJECT_VERSION_MINOR})
             set(previous "${current}")
             set(current "Qt_${PROJECT_VERSION_MAJOR}.${minor_version}")
@@ -49,6 +50,7 @@ function(qt_internal_add_linker_version_
                 string(APPEND contents "${current} {} ${previous};\n")
             endif()
         endforeach()
+        endif()
 
         set(infile "${CMAKE_CURRENT_BINARY_DIR}/${target}.version.in")
         set(outfile "${CMAKE_CURRENT_BINARY_DIR}/${target}.version")
