diff -up qtbase-everywhere-src-6.2.3/src/gui/kernel/qguiapplication.cpp.2~ qtbase-everywhere-src-6.2.3/src/gui/kernel/qguiapplication.cpp
--- qtbase-everywhere-src-6.2.3/src/gui/kernel/qguiapplication.cpp.2~	2022-01-28 23:32:06.594899424 +0100
+++ qtbase-everywhere-src-6.2.3/src/gui/kernel/qguiapplication.cpp	2022-01-28 23:35:15.544108103 +0100
@@ -1488,6 +1488,9 @@ void QGuiApplicationPrivate::createPlatf
     Q_UNUSED(platformExplicitlySelected);
 #endif
 
+    if(platformPluginPath.isEmpty())
+        platformPluginPath = QStringLiteral("@QTDIR@/plugins");
+
     init_platform(QLatin1String(platformName), platformPluginPath, platformThemeName, argc, argv);
 
     if (!icon.isEmpty())
