diff -up qtbase-everywhere-src-6.4.0-beta2/src/plugins/platforms/xcb/nativepainting/qtessellator.cpp.omv~ qtbase-everywhere-src-6.4.0-beta2/src/plugins/platforms/xcb/nativepainting/qtessellator.cpp
--- qtbase-everywhere-src-6.4.0-beta2/src/plugins/platforms/xcb/nativepainting/qtessellator.cpp.omv~	2022-07-20 16:47:40.930087115 +0200
+++ qtbase-everywhere-src-6.4.0-beta2/src/plugins/platforms/xcb/nativepainting/qtessellator.cpp	2022-07-20 16:47:56.361201590 +0200
@@ -5,6 +5,7 @@
 
 #include <QRect>
 #include <QList>
+#include <QMap>
 #include <QDebug>
 
 #include <qmath.h>
