diff -up qtbase-everywhere-src-6.7.0/src/corelib/tools/qspan.h.omv~ qtbase-everywhere-src-6.7.0/src/corelib/tools/qspan.h
--- qtbase-everywhere-src-6.7.0/src/corelib/tools/qspan.h.omv~	2024-04-18 00:30:57.556073100 +0200
+++ qtbase-everywhere-src-6.7.0/src/corelib/tools/qspan.h	2024-04-18 00:31:07.344169889 +0200
@@ -297,7 +297,7 @@ public:
         : QSpanBase(il.begin(), il.size())
     {}
 
-#if __cpp_lib_span
+#ifdef __cpp_lib_span
     template <typename S, size_t N, if_qualification_conversion<S> = true>
     Q_IMPLICIT constexpr QSpanBase(std::span<S, N> other) noexcept
         : QSpanBase(other.data(), other.size())
